src/nanoarrow/nanoarrow_device_cuda.c

```diff
@@ -157,6 +160,13 @@ static ArrowErrorCode ArrowDeviceCudaAllocateBuffer(struct ArrowDevice* device,
   return NANOARROW_OK;
 }
 
+static ArrowErrorCode ArrowDeviceCudaAllocateBuffer(struct ArrowDevice* device,
+                                                    struct ArrowBuffer* buffer,
+                                                    int64_t size_bytes) {
+  return ArrowDeviceCudaAllocateBufferAsync(device, buffer, size_bytes,
+                                            NANOARROW_CUDA_DEFAULT_STREAM);
+}
```

Comment:
I don't think this is safe to do. CUDA Streams can be created that are non-blocking with respect to the default stream where this could lead to races if someone assumes that `ArrowDeviceCudaAllocateBuffer` is synchronous and safe to use from any stream.

Additionally, the default stream can be controlled at compile time where it can be set to a per thread stream which would make this extra problematic.